tests/defer_regress/tests.py

```diff
@@ -238,3 +239,40 @@ def test_defer_annotate_select_related(self):
                  .defer('request1', 'request2', 'request3', 'request4')),
             list
         )
+
+
+class DeferDeletionSignalsTests(TestCase):
+    senders = [Item, Proxy]
+
+    def setUp(self):
+        self.clear_received_senders()
+        for sender in self.senders:
+            models.signals.pre_delete.connect(self.pre_delete_receiver, sender)
+            models.signals.post_delete.connect(self.post_delete_receiver, sender)
+
+    def tearDown(self):
+        for sender in self.senders:
+            models.signals.pre_delete.disconnect(self.pre_delete_receiver, sender)
+            models.signals.post_delete.disconnect(self.post_delete_receiver, sender)
+
+    def clear_received_senders(self):
+        self.pre_delete_senders = []
+        self.post_delete_senders = []
+
+    def pre_delete_receiver(self, sender, **kwargs):
+        self.pre_delete_senders.append(sender)
+
+    def post_delete_receiver(self, sender, **kwargs):
+        self.post_delete_senders.append(sender)
+
+    def test_delete_defered_model(self):
+        item = Item.objects.create(value=1)
+        Item.objects.only('value').get(pk=item.pk).delete()
+        self.assertEqual(self.pre_delete_senders, [Item])
+        self.assertEqual(self.post_delete_senders, [Item])
+
+        self.clear_received_senders()
```

Comment:
Receivers can't be connected in `setUpClass()` as they need to store the received senders on the test instance itself. I split the test in two methods and used `setUpTestData()`